CDS/js/youtube/youtube-auth.js

```diff
@@ -0,0 +1,97 @@
+var googleapis = require('googleapis');
+var OAuth2 = googleapis.auth.OAuth2;
+var fs = require('fs');
+var Promise = require('promise');
+var pem = require('pem');
+var Q = require('q');
+
+const SECRET_KEY_FILE_PATH = './privatekey.pem';
+const SCOPES = [
+    "https://www.googleapis.com/auth/youtube.force-ssl",
+    "https://www.googleapis.com/auth/youtube",
+    "https://www.googleapis.com/auth/youtube.readonly",
+    "https://www.googleapis.com/auth/youtube.upload"
+];
+
+function getCredentials() {
+
+    return new Promise((fulfill, reject) => {
+
+        const credentialsFile = process.env.client_secrets;
```

Comment:
just a thought - do we want to allow string substitutions in this?  this would allow us to e.g., have seperate credentials file paths in different environments using e.g.<client_secrets>youtube_{config:youtube_acct}_creds.json</client_secrets>